core/src/com/itemshop/character/walking/PathingSystem.java

```diff
@@ -138,6 +138,9 @@ protected void processEntity(Entity entity) {
 				// Get the facing direction of this entity.
 				Direction facingDirection = facingDirectionMapper.get(entity).direction;
 
+				// We are exiting the current walkable tile.
+				getWalkableTileComponentAtPosition(position.x, position.y).onExit.perform();
```

Comment:
Ah I see further down they are given an empty value by default. Possibly null would be better, if you want to clear this after assigning it you end up having to assign a new empty block of code